### ISSUE TITLE
Swapped livenessProbe and readinessProbe endpoints in 1.4.2

### ISSUE THREAD

**Issue by flozzone** [operator-framework/operator-sdk#4541]
#### What did you do?

Generated a ansible based operator with operator-sdk 1.4.2

#### What did you expect to see?

'config/manager/manager.yaml' containing the `livenessProbe` and `readinessProbe` from the upgrade guide:
https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.4.0/#for-ansible-based-operators-add-liveness-and-readiness-probe
```
  livenessProbe:
    httpGet:
      path: /healthz
      port: 6789
    initialDelaySeconds: 15
    periodSeconds: 20
  readinessProbe:
    httpGet:
      path: /readyz
      port: 6789
    initialDelaySeconds: 5
    periodSeconds: 10
```

#### What did you see instead? Under which circumstances?

`livenessProbe` using `/readyz` endpoint and `readinessProbe` using `/healthz` endpoint. Should be the other way around.
```
  livenessProbe:
    httpGet:
      path: /readyz
      port: 6789
    initialDelaySeconds: 15
    periodSeconds: 20
  readinessProbe:
    httpGet:
      path: /healthz
      port: 6789
    initialDelaySeconds: 5
    periodSeconds: 10
```

#### Environment

**Operator type:**

/language ansible
/language helm

**Kubernetes cluster type:**
Openshift
Server Version: 4.6.15
Kubernetes Version: v1.19.0+1833054

`$ operator-sdk version`

operator-sdk version: "v1.4.2", commit: "4b083393be65589358b3e0416573df04f4ae8d9b", kubernetes version: "1.19.4", go version: "go1.15.5", GOOS: "linux", GOARCH: "amd64"

`$ kubectl version`

Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0+1833054", GitCommit:"183305424c91636d683dfb4a379204c313976be2", GitTreeState:"clean", BuildDate:"2021-01-22T14:47:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

#### Possible Solution

Swap the two endpoints

#### Additional context

Added a code review to PR #4326 regarding this.

Please see also the wrong path to the manager resource file in the upgrade guide https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.4.0/#for-ansible-based-operators-add-liveness-and-readiness-probe
 `config/default/manager/manager.yaml` -> 'config/manager/manager.yaml'

**Comment by flozzone** [comment]
Please add label `language/helm` since helm operators seem also to be affected.

**Comment by camilamacedo86** [comment]
WDYT about collab with the PR to fix the docs?

**Comment by flozzone** [comment]
Sure. Should I create a separate issue for the documentation changes?

**Comment by camilamacedo86** [comment]
HI @flozzone, 

The following is how it is Golang and how it should be for Ansible/Helm:

livenessProbe: ->  /healthz
readinessProbe: ->  /readyz
          
By checking the Anible  [here](https://github.com/operator-framework/operator-sdk/blob/master/testdata/ansible/memcached-operator/config/manager/manager.yaml#L34) and [Helm](https://github.com/operator-framework/operator-sdk/blob/master/testdata/helm/memcached-operator/config/manager/manager.yaml#L31-L42) we can see that it is inverted : 

 livenessProbe: ->  /readyz
readinessProbe: ->  /healthz

So, we need to; 
- change the templates to fix that
- create a changelog fragment with the steps to fix it for projects already scaffolded

Then, we might need to do a pr against the released branches.

WDYT? Could you help on that?

**Comment by flozzone** [comment]
Seems to be quite straight forward. I've already marked the respective parts of the code in #4326. But I will read through your code committing docs before, since this would be my first contribution.

**Comment by flozzone** [comment]
@camilamacedo86 these are the two commits. I've left them apart since they wouldn't fit well in one (and used the same issue number).

It looks like testdata is generated (https://github.com/operator-framework/operator-sdk/blob/master/hack/generate/samples/internal/ansible/memcached.go#L129). But I have no running dev env for this project right now. I've just fixed the testdata manually.

**Comment by flozzone** [comment]
How should I proceed with the separated commits? squash or create a separate issue for the docs issue?

**Comment by estroz** [comment]
@flozzone make 2 separate PR's. The one containing the name change should have a ["bugfix" changelog fragment](https://github.com/operator-framework/operator-sdk/blob/master/changelog/fragments/00-template.yaml).

**Comment by flozzone** [comment]
Is it ok, having two PRs referencing the same issue? #4545 and #4546

**Comment by estroz** [comment]
Fixed by #4546
